core/src/test/scala/org/apache/spark/rdd/RDDSuite.scala

```diff
@@ -154,6 +154,13 @@ class RDDSuite extends SparkFunSuite with SharedSparkContext {
     }
   }
 
+  test("SPARK-23778: empty RDD in union should not produce a UnionRDD") {
```

Comment:
Have we tested when all input RDDs are empty?

Reply:
When all RDDs are empty we are returning a `UnionRDD`. Though in this case it is not a big issue, since a shuffle of an empty RDD is not an issue.

Reply:
can we add a test? just make sure we are safe when `UnionRDD.rdds` is Nil

Reply:
added, thanks.